server/src/models/RoomModel.mjs

```diff
@@ -26,27 +26,29 @@ export default class RoomModel {
     return result.recordset;
   }
 
-  static async getRoomByStatus(Status) {
-    const result = await connection
-      .request()
-      .input('Status', Status)
-      .query('SELECT * FROM Room WHERE Status = @Status');
+  static async getRoomByStatus(IsAvailable) {
+    const result = await connection.request().input('IsAvailable', IsAvailable)
+      .query(`SELECT r.RoomID as Number, t.Type, t.Max_Occupancy as MaxOccupancy, t.Min_Customer_for_Surcharge as BaseCustomers, t.Price, r.ImgUrl, t.Surcharge_Rate as SurchargeRate
+              FROM ROOM r join ROOMTYPE t on r.Type = t.Type
+              WHERE IsAvailable = @IsAvailable`);
     return result.recordset;
   }
 
-  static async getRoomByTypeAndStatus(Type, Status) {
+  static async getRoomByTypeAndStatus(Type, IsAvailable) {
     const result = await connection
       .request()
       .input('Type', Type)
-      .input('Status', Status)
-      .query('SELECT * FROM Room WHERE Type = @Type AND Status = @Status');
+      .input('IsAvailable', IsAvailable)
+      .query(
+        'SELECT * FROM Room WHERE Type = @Type AND IsAvailable = @IsAvailable'
+      );
     return result.recordset;
   }
 
   static async createRoom(
```

Comment:
Need to handle cases where user accidentally create an **already-existed room**; when this happens, need to have proper notification.
Currently, the page just displays 'Room is being saved...' (can still go to other sections)
![image](https://github.com/user-attachments/assets/5822ba2e-8137-407b-8801-fc7ad3ad08f6)

_Room 101 already exists, has type A_

Reply:
- Ok thanks for pointing out this silly but important bug
- I mentioned it in Issue #17 and also committed the fix in **main** branch
- Since it was a simple process. I was lazy and didn't create another branch haha XD